### DISCUSSION TITLE
Import containers

### DISCUSSION THREAD

**Discussion by algorlab** [terrastruct/d2#323] [Feature Requests]
Can I create a component in a d2 file (e.g. a `Container`) and then import that component into another .d2 file?

**Comment by alixander** [comment]
not currently. this is a TODO, and allows D2 diagrams to be way more composable. ty for starting the discussion to track

**Reply by alixander** [reply]
update, this is done

**Comment by muteboy** [comment]
I was looking to see if I could use an external file to define all the classes I need, and I think this would be it.

**Reply by muteboy** [reply]
The current import functionality is useful, but why can't I use it to define a set of classes?

classes.d2:
```
classes: {
  state: {
    shape: parallelogram
    height: 40
    style: {
      fill: limegreen
    }
 etc
```

then in my main D2 file:
`...@classes`

**Reply by alixander** [reply]
This is a common pattern that works. 

What error are you seeing?

**Reply by muteboy** [reply]
I'm using the VS Code plugin v0.8.8. In classes.d2 I have:

```
# CLASS DEFINITIONS
classes: {
  state: {
    shape: parallelogram
    height: 40
    style: {
      fill: limegreen
    }
  ...etc etc...
}

# GLOBAL CLASS SETTINGS
State*.class: state
**.State*.class: state
...etc etc...
```

In my main file I have:
`...@classes`

The classes aren't applied in the main file - they show up as grey rectangles.

However, if I put the `GLOBAL CLASS SETTINGS` in the main file, then the classes are applied correctly.
